### PR TITLE
chore: wrap metric with substrafl add_metric function

### PR DESCRIPTION
**Pull request by ThibaultFy**
# Companion PR

- https://github.com/Substra/substrafl/pull/47

### CODE REVIEW

File: substrafl_examples/strategies_examples/plot_substrafl_fedavg.py [hunk 1]
```diff
@@ -167,46 +167,31 @@
...
-archive_path = objective.file
-with zipfile.ZipFile(archive_path, "w") as z:
-    for filepath in METRICS_DOCKERFILE_FILES:
-        z.write(filepath, arcname=filepath.name)
+    def load_predictions(self, path):
+        return np.load(path)
 
-metric_key = clients[ALGO_ORG_ID].add_algo(objective)
+
+metric_key = register_metric(
```
Comment (ThibaultFy) [inline substrafl_examples/strategies_examples/plot_substrafl_fedavg.py:189]: rename to add_metric

File: docs/source/substrafl_doc/substrafl_overview.rst [hunk 2]
```diff
@@ -60,6 +61,12 @@ The evaluation strategy specifies how and when the model is tested. More specifi
 * on which test data the model is tested
 * at which rounds the model is tested
 
+Metric
+^^^^^^
+
+A metric is a function that computes a performance between data and predictions loaded from the organizations.
```
Comment (RomainGoussault) [inline docs/source/substrafl_doc/substrafl_overview.rst:67]:
```suggestion
A metric is a function that computes a performance by comparing label (ground truth) and model's predictions.
```

File: substrafl_examples/get_started/plot_substrafl_torch_fedavg.py [hunk 3]
```diff
@@ -148,73 +148,61 @@
...
     test_datasample_keys[org_id] = client.add_data_sample(data_sample)
 
+
 # %%
-# Metrics registration
-# ====================
+# Metric registration
+# ===================
 #
 # A metric is an algorithm used to compute the score of predictions on one or several
```
Comment (RomainGoussault) [inline substrafl_examples/get_started/plot_substrafl_torch_fedavg.py:168]:
```suggestion
# A metric is a function used to compute the score of predictions on one or several
```

File: substrafl_examples/get_started/plot_substrafl_torch_fedavg.py [hunk 4]
```diff
@@ -148,73 +148,61 @@
...
-# - a `Dockerfile <https://docs.docker.com/engine/reference/builder/>`__ to specify the required dependencies of the
-#   **Python scripts**
+# To add a metric, you need to define a function that computes and return a performance
+# from the datasamples (as returned by the opener) and the predictions_path (to be loaded within the function).
+#
+# When using a Torch SubstraFL algorithm, the predictions are saved in the `predict` function under the numpy format
+# so that you can simply load them using `np.load`.
+#
+# After defining the metrics dependencies and permissions, we use the add_metric function to register the metric on
+# the platform. This metric will be used on the test datasamples to evaluate the model performances.
```
Comment (RomainGoussault) [inline substrafl_examples/get_started/plot_substrafl_torch_fedavg.py:178]:
```suggestion
# After defining the metrics dependencies and permissions, we use the add_metric function to register the metric.
# This metric will be used on the test datasamples to evaluate the model performances.
```

**Review by RomainGoussault — COMMENTED** [review]
So much simpler now with add_metric 🤩

File: substrafl_examples/get_started/plot_substrafl_torch_fedavg.py (hunk 4 above)
Comment (ThibaultFy) [reply substrafl_examples/get_started/plot_substrafl_torch_fedavg.py]: done

**Review by RomainGoussault — APPROVED** [review]
LGTM!

**Pull request merged**